src/Autofac.Integration.AspNetCore.Multitenant/AutofacServiceScope.cs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Autofac Project. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Autofac;
+using Autofac.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection;
+
+/// <summary>
+/// Autofac implementation of the ASP.NET Core <see cref="IServiceScope"/>.
+/// </summary>
+/// <seealso cref="IServiceScope" />
+internal class AutofacServiceScope : IServiceScope, IAsyncDisposable
```

Comment:
If it turns out we need to create service scopes directly, it might be better to make this public in the A.E.DI library.

But I'm wondering if we can make use of the [AutofacChildLifetimeScopeServiceProviderFactory](https://github.com/autofac/Autofac.Extensions.DependencyInjection/blob/develop/src/Autofac.Extensions.DependencyInjection/AutofacChildLifetimeScopeServiceProviderFactory.cs) here, where each tenant has their own instance of this child scope provider factory. That looks a lot like what you've started building here.

Reply:
Hmmm. I may have spoken a little too soon. Let me work on a larger comment, but this is still along the lines of what I'm thinking.